From 02d0e45b311d224c70a6c8af99d5eafe715e5388 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 27 Nov 2013 13:45:16 +0800
Subject: [PATCH 08/11] ENGR00289825 Make simulated battery status report
 based on sys.emulated.battery

Fake a battery status based on sys.emulated.battery property setting
Boards without battery support should have this property set, due to some
function like data encryption need to check battery status.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 healthd/BatteryMonitor.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/healthd/BatteryMonitor.cpp b/healthd/BatteryMonitor.cpp
index 688c7ff..df29e63 100644
--- a/healthd/BatteryMonitor.cpp
+++ b/healthd/BatteryMonitor.cpp
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
+/*
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
+ */
+
 #define LOG_TAG "healthd"
 
 #include "healthd.h"
@@ -28,6 +32,7 @@
 #include <unistd.h>
 #include <batteryservice/BatteryService.h>
 #include <cutils/klog.h>
+#include <cutils/properties.h>
 #include <utils/String8.h>
 #include <utils/Vector.h>
 
@@ -265,6 +270,15 @@ bool BatteryMonitor::update(void) {
                   props.chargerWirelessOnline ? "w" : "");
     }
 
+    char prop[PROP_VALUE_MAX];
+    // always report AC plug-in and capacity 100% if emulated.battery is set to 1
+    property_get("sys.emulated.battery", prop, "0");
+    if (!strcmp(prop, "1")){
+        props.chargerAcOnline = true;
+        props.batteryLevel = 100;
+        props.chargerWirelessOnline = false;
+    }
+
     if (mBatteryPropertiesRegistrar != NULL)
         mBatteryPropertiesRegistrar->notifyListeners(props);
 
-- 
1.8.0

